internal/api/otp.go

```diff
@@ -198,17 +199,21 @@ func (a *API) SmsOtp(w http.ResponseWriter, r *http.Request) error {
 		if terr != nil {
 			return badRequestError("Error sending sms: %v", terr)
 		}
-		if err := a.sendPhoneConfirmation(ctx, tx, user, params.Phone, phoneConfirmationOtp, smsProvider, params.Channel); err != nil {
-			return badRequestError("Error sending sms otp: %v", err)
+		mID, serr := a.sendPhoneConfirmation(ctx, tx, user, params.Phone, phoneConfirmationOtp, smsProvider, params.Channel)
+		if serr != nil {
+			return badRequestError("Error sending sms OTP: %v", err)
 		}
+		messageID = mID
 		return nil
 	})
 
 	if err != nil {
 		return err
 	}
 
-	return sendJSON(w, http.StatusOK, make(map[string]string))
+	return sendJSON(w, http.StatusOK, map[string]any{
+		"message_id": messageID,
+	})
```

Comment:
can we format this into a struct so we can omit the `message_id` if it isn't present?

for e.g. 
```go
type SmsOtpResponse struct {
  MessageID string `json:"message_id,omitempty"`
}
```